Sophos/sophos_module/trigger_sophos_xdr_query.py

```diff
@@ -198,11 +198,12 @@ def getting_results(self, pagination: int) -> None:
                     ).json()
 
                     next_page_items = response_next_page.get("items", [])
+                    next_messages = [orjson.dumps(message).decode("utf-8") for message in next_page_items]
                     self._observe_items_events_lag(next_page_items)
-                    grouped_data.extend(next_page_items)
+                    grouped_data.extend(next_messages)
                     self.log(message=f"Sending other batches of {len(grouped_data)} messages", level="info")
                     OUTCOMING_EVENTS.labels(intake_key=self.configuration.intake_key).inc(len(grouped_data))
-                    self.push_events_to_intakes(events=grouped_data)
+                    self.push_events_to_intakes(events=next_messages)
```

Comment:
It should not stay `grouped_data` here?
You update line 201 in order that the messages pushed in `grouped_data` are JSON document.
Unless `grouped_data` is useless and can be replaced by `next_messages` (it seems to be the case).

```suggestion
                    self.push_events_to_intakes(events=grouped_data)
```

Reply:
Yes normally it should be `next_messages `. So i will delete `grouped_data`.